### PR TITLE
#218 chore: added (email) notifications

### PR DESCRIPTION
**Pull request by zsinnema**

### CODE REVIEW

File: applications/workspaces/server/workspaces/service/notification/adapters/email_adapter.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def notify(self, context):
+        logger.info(f"Sending notification email to {self.email_to}")
+        msg = EmailMessage()
+        msg['Subject'] = self.subject
+        msg['From'] = self.email_from
+        msg['To'] = self.email_to
+        content = self.render_content(context)
+        msg.set_content(content, subtype='html')
+
+        smtp = smtplib.SMTP(EMAIL_HOST, EMAIL_PORT)
```
Comment (filippomc) [inline applications/workspaces/server/workspaces/service/notification/adapters/email_adapter.py:40]: anything from here may easily fail and may break the main routine.

File: applications/workspaces/server/workspaces/repository/model_repository.py [hunk 2]
```diff
@@ -115,6 +116,21 @@ def post_commit(self, workspace):
                 db.session.commit()
         return workspace
 
+    def post(self, body):
+        obj = super().post(body)
+        if isinstance(obj, tuple):
+            # http return code in result, this means an error, return it
+            return obj
+        # no error, send notification
+        notify(
```
Comment (filippomc) [inline applications/workspaces/server/workspaces/repository/model_repository.py:125]: Better be catched and logged

**Review by filippomc — CHANGES_REQUESTED** [review]
needs some more work on exception handling

**Comment by zsinnema** [comment]
replaced by pr #231

**Pull request closed without merging**